### PR TITLE
feat: [FFM-2087]: Added developer limit banner and plan enforcement tooltip to access control page

### PR DESCRIPTION
**Pull request by chrisblakely01**
##### Summary:

- Adds the plan enforcement developer limit banner to the access control pages
- Adds the plan enforcement tooltip to the "Feature Flag Management" option in the "New User" > "Roles" dropdown 

##### Changes:

- Added `DeveloperLimitBanner` and tests to `rbac` module
- Move `user role` dropdown from `RoleAssignmentForm` to its own component `NewUserRoleDropdown` and added tests (This made it easier to isolate/write cleaner tests for my change"

##### Jira Links:

[FFM-2087](https://harness.atlassian.net/jira/software/c/projects/FFM/boards/98?modal=detail&selectedIssue=FFM-2087)

#####  Figma Links:

https://www.figma.com/file/KBWLXtXUnh6leVq0l7OzNG/Product-Led-Growth?node-id=570%3A68763

##### Screenshots:

If `FEATURE_ENFORCEMENT_ENABLED` is disabled, the functionality remains the same:

Screenshot 2022-01-19 at 10 15 57

IF `FEATURE_ENFORCEMENT_ENABLED` is enabled, and the account has reached its FF free plan limits, the "Feature Flag Management" option is disabled, and a tooltip is shown:

Screenshot 2022-01-19 at 10 08 31

We also show the developer limit banner in this case:

Screenshot 2022-01-24 at 11 18 49

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { TestWrapper } from '@common/utils/testUtils'
+// eslint-disable-next-line no-restricted-imports
```
Comment (chrisblakely01) [inline src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx:11]: This isn't ideal but this is the platform hook I need to use. Moving this component to `common` results in the same thing linting error - Suggestions welcome

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.module.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+.selectPopover {
```
Comment (chrisblakely01) [inline src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.module.scss:8]: FYI This was moved from the existing stylesheet

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+const NewUserRoleDropdown = (props: NewUserRoleDropdownProps): ReactElement => {
+  const { value, roles, handleChange } = props
+  const { getString } = useStrings()
+
+  const isFeatureEnforcementEnabled = useFeatureFlag(FeatureFlag.FEATURE_ENFORCEMENT_ENABLED)
+
+  return (
+    <Select
+      itemRenderer={(item: SelectOption, { handleClick }): React.ReactElement => {
+        if (isFeatureEnforcementEnabled && item.value === '_feature_flag_manage_role') {
```
Comment (chrisblakely01) [inline src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx:35]: This new functionality should only apply to `_feature_flag_manage_role` - is this a safe check? is `_feature_flag_manage_role` consistent across the environments?

File: src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ */
+
+import React, { ReactElement } from 'react'
+import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
+import FeatureWarningSubscriptionInfoBanner from '@common/components/FeatureWarning/FeatureWarningSubscriptionInfoBanner'
+import FeatureWarningSubscriptionUpgradeBanner from '@common/components/FeatureWarning/FeatureWarningSubscriptionUpgradeBanner'
+import { useLicenseStore } from 'framework/LicenseStore/LicenseStoreContext'
+import { useStrings } from 'framework/strings'
+// eslint-disable-next-line no-restricted-imports
+import { useGetUsageAndLimit } from '@auth-settings/hooks/useGetUsageAndLimit'
```
Comment (reetika04) [inline src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx:15]: Please move useGetUsageAndLimit to common instead.

File: src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+import * as useGetUsageAndLimitMock from '@auth-settings/hooks/useGetUsageAndLimit'
+import * as licenseStoreContextMock from 'framework/LicenseStore/LicenseStoreContext'
+import * as useFeatureFlagMock from '@common/hooks/useFeatureFlag'
+import DeveloperLimitBanner from '../DeveloperLimitBanner'
+
+const renderComponent = (): void => {
+  render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
```
Comment (reetika04) [inline src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx:20]: Please use routes

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+              text={item.value}
+              featuresProps={{
+                featuresRequest: {
+                  featureNames: [FeatureIdentifier.MAUS]
+                }
+              }}
+            />
+          )
+        } else {
+          return <Menu.Item key={item.value as string} text={item.value} onClick={handleClick} />
```
Comment (reetika04) [inline src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx:49]:
```suggestion
          return <Menu.Item key={item.value as string} text={item.label} onClick={handleClick} />
```

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+  return (
+    <Select
+      itemRenderer={(item: SelectOption, { handleClick }): React.ReactElement => {
+        if (isFeatureEnforcementEnabled && item.value === '_feature_flag_manage_role') {
+          return (
+            <RbacMenuItem
+              onClick={handleClick}
+              key={item.value as string}
+              text={item.value}
```
Comment (reetika04) [inline src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx:40]:
```suggestion
              text={item.label}
```

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx (hunk 3 above)
Comment (reetika04) [reply src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx:35]: This is not a safe check, Any managed role such as _feature_flag_manage_role could be replicated as a custom role. Account/Project Admin is also super set of  FF Manage Role.

File: src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+const DeveloperLimitBanner = (): ReactElement => {
+  const { getString } = useStrings()
+  const { limitData, usageData } = useGetUsageAndLimit(ModuleName.CF)
+
+  const isFeatureEnforcementEnabled = useFeatureFlag(FeatureFlag.FEATURE_ENFORCEMENT_ENABLED)
+
+  const license = useLicenseStore()
+
+  const isPaidTeamOrEnterpriseSubscription =
```
Comment (reetika04) [inline src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx:28]: Can we check why this is specific to FF?

**Review by reetika04 — CHANGES_REQUESTED** [review]

File: src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx (hunk 4 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx]: moved

File: src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx (hunk 8 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/DeveloperLimitBanner/DeveloperLimitBanner.tsx:27]: See answers in comments here: https://harness.atlassian.net/browse/FFM-2087

File: src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx (hunk 1 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx]: fixed as per comments above

File: src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx (hunk 5 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/DeveloperLimitBanner/__tests__/DeveloperLimitBanner.test.tsx]: fixed

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx (hunk 3 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx]: See answers in comments here: https://harness.atlassian.net/browse/FFM-2087

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx (hunk 7 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx]: fixed

File: src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx (hunk 6 above)
Comment (chrisblakely01) [reply src/modules/20-rbac/components/NewUserRoleDropdown/NewUserRoleDropdown.tsx]: fixed

**Review by reetika04 — APPROVED** [review]
There are a few other open questions on this PR listed here https://harness.atlassian.net/browse/FFM-2087

**Review by willd0yle — APPROVED** [review]

**Pull request merged**